kqueue.go

```diff
@@ -119,6 +121,13 @@ func (w *Watcher) Remove(name string) error {
 	w.mu.Lock()
 	isDir := w.paths[watchfd].isDir
 	delete(w.watches, name)
+
+	parentName := filepath.Dir(name)
+	delete(w.watchesByDir[parentName], watchfd)
+	if len(w.watchesByDir[parentName]) == 0 {
+		delete(w.watchesByDir, parentName)
```

Comment:
Still not quite understanding the point of this. If the parentName has no child watches, why do we want to delete it from watchesByDir?

Reply:
_shrug_, i don't mind removing it.

Reply:
Ah I finally got the purpose of this. We also need to check if the watchesByDir is empty for the current name and if so, to delete it there.

Reply:
ok, i'll add it back?